frontend-public/src/components/dashboard/mine/variances/VarianceTable.js

```diff
@@ -0,0 +1,160 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Table } from "antd";
+import CustomPropTypes from "@/customPropTypes";
+import { formatDate } from "@/utils/helpers";
+import { RED_CLOCK } from "@/constants/assets";
+import downloadFileFromDocumentManager from "@/utils/actionlessNetworkCalls";
+import * as Strings from "@/constants/strings";
+import { COLOR } from "@/constants/styles";
+
+const { errorRed } = COLOR;
+
+const propTypes = {
+  variances: PropTypes.arrayOf(CustomPropTypes.variance).isRequired,
+  complianceCodesHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  varianceStatusOptionsHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  isApplication: PropTypes.bool,
+};
+
+const defaultProps = {
+  isApplication: false,
+};
+
+const errorStyle = (isOverdue) => (isOverdue ? { color: errorRed } : {});
+const transformRowData = (variances, codeHash, statusHash) =>
+  variances.map((variance) => ({
+    key: variance.variance_guid,
+    variance,
+    status: statusHash[variance.variance_application_status_code],
+    compliance_article_id: codeHash[variance.compliance_article_id] || Strings.EMPTY_FIELD,
+    expiry_date: formatDate(variance.expiry_date) || Strings.EMPTY_FIELD,
+    issue_date: formatDate(variance.issue_date) || Strings.EMPTY_FIELD,
+    note: variance.note,
+    received_date: formatDate(variance.received_date) || Strings.EMPTY_FIELD,
+    isOverdue: Date.parse(variance.expiry_date) < new Date(),
```

Comment:
`variance.expiry_date` is nullable. I'd recommend:
```javascript
expiry_date: variance.expiry_date && Date.parse(variance.expiry_date) < new Date(),
isOverdue: variance.expiry_date && Date.parse(variance.expiry_date) < new Date(),
```

I've updated CORE in this way on `MDS-1859-no-9999`

Reply:
updated!